fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1062,6 +1056,329 @@ Future<Void> managementClusterRemoveTenantFromGroup(Transaction tr,
 	return Void();
 }
 
+template <class DB>
+struct RestoreClusterImpl {
+	MetaclusterOperationContext<DB> ctx;
+
+	// Initialization parameters
+	ClusterName clusterName;
+	ClusterConnectionString connectionString;
+	DataClusterEntry clusterEntry;
+	AddNewTenants addNewTenants;
+	RemoveMissingTenants removeMissingTenants;
+
+	// Tenant list from data and management clusters.
+	std::unordered_map<TenantName, TenantMapEntry> dataClusterTenantMap;
+	std::unordered_map<uint64_t, TenantName> dataClusterTenantIdIndex;
+	std::unordered_map<TenantName, TenantMapEntry> mgmtClusterTenantMap;
+	std::unordered_map<uint64_t, TenantName> mgmtClusterTenantIdIndex;
+	std::unordered_set<TenantNameRef> mgmtTenantSet;
```

Comment:
I think we could similarly build all three of these data structures from the single read across the management cluster. Although possible we don't actually need all of the tenants across the entire metacluster, but instead just the ones from the target cluster? That may change with different restore types.